src/librustc_lint/non_ascii_idents.rs

```diff
@@ -13,9 +14,50 @@ declare_lint! {
     "detects uncommon Unicode codepoints in identifiers"
 }
 
-declare_lint_pass!(NonAsciiIdents => [NON_ASCII_IDENTS, UNCOMMON_CODEPOINTS]);
+declare_lint! {
+    pub CONFUSABLE_IDENTS,
+    Warn,
+    "detects uncommon Unicode codepoints in identifiers"
+}
+
+declare_lint_pass!(NonAsciiIdents => [NON_ASCII_IDENTS, UNCOMMON_CODEPOINTS, CONFUSABLE_IDENTS]);
+
+fn calc_skeleton(symbol_str: &str) -> String {
+    use unicode_security::confusable_detection::skeleton;
+    skeleton(symbol_str).collect()
+}
 
 impl EarlyLintPass for NonAsciiIdents {
+    fn check_crate(&mut self, cx: &EarlyContext<'_>, _: &ast::Crate) {
+        let mut symbol_strs_and_spans = Vec::new();
+        let symbols = cx.sess.parse_sess.symbol_gallery.symbols.lock();
+        for (symbol, sp) in symbols.iter() {
+            let symbol_str = symbol.as_str();
+            symbol_strs_and_spans.push((symbol_str, *sp));
```

Comment:
We should be able to preallocate this vector (with_capacity).

Reply:
Good idea, will do.